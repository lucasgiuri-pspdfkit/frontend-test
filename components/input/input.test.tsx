import Input from "./input";
import { render, screen, fireEvent } from "@testing-library/react";

const availableTypes = ["text", "password", "email"];

describe("Input", () => {
  it("Snapshot with empty placeholder and empty value", () => {
    const onChange = jest.fn();
    const { container } = render(
      <Input placeholder="" value="" onChange={onChange} />
    );
    expect(container).toMatchSnapshot();
  });

  availableTypes.forEach((type) => {
    it(`Snapshot with type ${type}`, () => {
      const onChange = jest.fn();
      const { container } = render(
        <Input
          placeholder="placeholder test"
          value="value test"
          onChange={onChange}
          type={type}
        />
      );
      expect(container).toMatchSnapshot();
    });
  });

  it("handles onChange event", () => {
    const handleChange = jest.fn();
    render(
      <Input
        placeholder="placeholder test"
        value="value test"
        onChange={handleChange}
      />
    );
    const elm = screen.getByPlaceholderText("placeholder test");
    fireEvent.change(elm, {
      target: { value: "new value" },
    });
    expect(handleChange).toHaveBeenCalledTimes(1);
    expect(handleChange).toHaveBeenCalledWith("new value");
  });
});
