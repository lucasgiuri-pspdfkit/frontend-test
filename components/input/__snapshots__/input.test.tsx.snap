// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshot with empty placeholder and empty value 1`] = `
<div>
  <input
    class="w-full px-4 py-2 border border-gray-100 focus:ring-0 outline-gray-200"
    placeholder=""
    type="text"
    value=""
  />
</div>
`;

exports[`Input Snapshot with type email 1`] = `
<div>
  <input
    class="w-full px-4 py-2 border border-gray-100 focus:ring-0 outline-gray-200"
    placeholder="placeholder test"
    type="email"
    value="value test"
  />
</div>
`;

exports[`Input Snapshot with type password 1`] = `
<div>
  <input
    class="w-full px-4 py-2 border border-gray-100 focus:ring-0 outline-gray-200"
    placeholder="placeholder test"
    type="password"
    value="value test"
  />
</div>
`;

exports[`Input Snapshot with type text 1`] = `
<div>
  <input
    class="w-full px-4 py-2 border border-gray-100 focus:ring-0 outline-gray-200"
    placeholder="placeholder test"
    type="text"
    value="value test"
  />
</div>
`;
